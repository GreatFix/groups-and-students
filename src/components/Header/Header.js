import React from 'react'
import classes from './Header.module.css'
import PropTypes from 'prop-types'

const Header = (props) => {
  return <div className={`${classes.Header} ${props.className}`}>{props.children}</div>
}

Header.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired,
}

Header.defaultProps = {
  className: '',
  children: null,
}

export default Header
