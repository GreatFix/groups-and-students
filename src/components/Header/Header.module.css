.Header {
  font-size: 1.2em;
  font-family: 'Lora', serif;
  text-align: center;
  padding: 5px;
  margin: 2px;
}
