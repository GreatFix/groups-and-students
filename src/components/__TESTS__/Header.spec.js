import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Header from '../Header/Header'

const TEST_TEXT = 'May the force be with you'
const TEST_CLASS = 'TEST_CLASS'

describe('Header', () => {
  test('renders a text', () => {
    const { getByText } = render(<Header>{TEST_TEXT}</Header>)
    expect(getByText(TEST_TEXT)).toBeInTheDocument()
  })

  test('gets a class', () => {
    const { container } = render(<Header className={TEST_CLASS}>{TEST_TEXT}</Header>)
    expect(container.querySelector('.Header')).toHaveClass(TEST_CLASS)
  })

  test('render properly', () => {
    const { container } = render(<Header>{TEST_TEXT}</Header>)
    expect(container).toMatchSnapshot()
  })
})
