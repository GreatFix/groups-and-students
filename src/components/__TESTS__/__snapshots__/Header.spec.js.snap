// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render properly 1`] = `
<div>
  <div
    class="Header "
  >
    May the force be with you
  </div>
</div>
`;
